sr_utilities/scripts/sr_utilities/hand_finder.py

```diff
@@ -104,7 +104,17 @@ def __init__(self, mapping, joint_prefix):
             if (rospy.get_time() - start_time > TIMEOUT_WAIT_FOR_PARAMS_IN_SECS):
                 rospy.logwarn("No robot_description found on parameter server."
                               "Joint names are loaded for 5 finger hand")
-                break
+            # concatenate all the joints with prefixes
+            for hand in mapping:
+                hand_joints = []
+                if hand in joint_prefix:
+                    for joint in joints:
+                        hand_joints.append(joint_prefix[hand] + joint)
+                else:
+                    rospy.logwarn("Cannot find serial " + hand +
+                                  "in joint_prefix parameters")
+                self.joints[mapping[hand]] = hand_joints
+            return
```

Comment:
I wonder if indentation is incorrect here? Removing `break` and adding unconditional `return` will make loop to run only once. Is is expected behaviour or I misunderstand how it works?
Also I wonder if `rospy.has_param("/robot_description")` is immediately true then this loop will never run. Is it ok without this initialisation?

Reply:
Oh yes, nice catch, the indentation is wrong. Fixed here 445cb2dce5f04ec77185cebc4e7abc6979e7dce3
It worked in testing because `rospy.has_param("/robot_description")` was immediately true. If everything is working properly then this code block should never be entered